app/models/user.rb

```diff
@@ -1,8 +1,22 @@
 class User < ApplicationRecord
+  enum role: %i[user admin]
+  after_initialize :set_default_role, if: :new_record?
+
+  has_many :articles
+
   # Include default devise modules. Others available are:
   # :confirmable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable, :lockable
 
-  has_many :articles
+  private
+
+  def set_default_role
+    self.role ||= :user
+  end
+
+  def create
+    @user = User.new(params[:user].permit(:id, :email, :password, :role))
```

Comment:
I think it doesn't work it because model has no `params`, this is responsibility of controller to `permit` params, while model could have validation

Reply:
removed create method